tests/testsuite/test.rs

```diff
@@ -135,6 +135,24 @@ fn cargo_test_overflow_checks() {
     p.process(&p.release_bin("foo")).with_stdout("").run();
 }
 
+#[test]
+fn cargo_test_quiet() {
+    let p = project()
+        .file("Cargo.toml", &basic_bin_manifest("foo"))
+        .file(
+            "src/main.rs",
+            r#"
+            fn main() {}
+            #[test] fn test_hello() {}
+        "#,
+        ).build();
+
+        p.cargo("test -q hello")
+            .with_stdout_contains("running 1 test")
```

Comment:
Shouldn't it _not_ contain this?  In fact, should the two assertions be the following?

```rust
.with_stdout("")
.with_stderr("")
```

Reply:
The output only contains `no of tests` being run plus the `test result`. It only doesn't show the actual tests being run. This is the same as doing
```
$ rustc --test unit-test.rs
$ ./unit-test --quiet
```
cc @dwijnand

Reply:
Right, that's what I thought.  libtest's help says:

```
    -q, --quiet         Display one character per test instead of one line.
                        Alias to --format=terse
```

So I think if we incorporate that information into `cargo test --help` and expand this test to test the full output of out and err, for me, we can make this change.